rtt/base/BufferLocked.hpp

```diff
@@ -130,10 +139,14 @@ namespace RTT
                 ++itl;
             }
             // this is in any case the number of elements taken from items.
+            size_t writtenSamples = itl - items.begin(); 
```

Comment:
The patch causes a bunch of new signed and unsigned integer comparison warnings to be emitted if RTT is compiled with all warnings enabled (see [Travis build #330](https://travis-ci.org/orocos-toolchain/rtt/builds/64374829), but the failure is due to resource constraints).

`size_t` -> `size_type` (== `int` as defined in BufferBase)?